core/app_telemetry_meter.cxx

```diff
@@ -0,0 +1,650 @@
+/* -*- Mode: C++; tab-width: 4; c-basic-offset: 4; indent-tabs-mode: nil -*- */
+/*
+ * Copyright 2024-Present Couchbase, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ * ANY KIND, either express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "app_telemetry_meter.hxx"
+
+#include "core/topology/configuration.hxx"
+
+#include <chrono>
+#include <spdlog/fmt/bundled/core.h>
+
+#include <algorithm>
+#include <array>
+#include <atomic>
+#include <map>
+#include <mutex>
+#include <utility>
+
+namespace couchbase::core
+{
+class byte_appender
+{
+public:
+  using iterator_category = std::output_iterator_tag;
+  using value_type = void;
+
+  explicit byte_appender(std::vector<std::byte>& output)
+    : buffer_{ output }
+  {
+  }
+
+  auto operator=(char ch) -> byte_appender&
+  {
+    buffer_.push_back(static_cast<std::byte>(ch));
+    return *this;
+  }
+
+  auto operator*() -> byte_appender&
+  {
+    return *this;
+  }
+
+  auto operator++() const -> byte_appender
+  {
+    return *this;
+  }
+
+  auto operator++(int) const -> byte_appender
+  {
+    return *this;
+  }
+
+private:
+  std::vector<std::byte>& buffer_;
+};
+} // namespace couchbase::core
+
+template<>
+struct fmt::detail::is_output_iterator<couchbase::core::byte_appender, char> : std::true_type {
+};
+
+namespace couchbase::core
+{
+namespace
+{
+
+class noop_app_telemetry_value_recorder : public app_telemetry_value_recorder
```

Comment:
Looks like all the histograms are missing the `agent` labels.

Also, it looks like the histogram buckets are missing `duration` in the name.
So they return:
`sdk_kv_mutation_nondurable_ms_bucket`
instead of:
`sdk_kv_mutation_nondurable_duration_ms_bucket`